torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1730,20 +1730,21 @@ def backward(ctx, *flat_args):
                     # Add the seed and offset to args
                     rng_args = CUDARngStateHelper.get_torch_state_as_tuple()
 
+                bw_tokens = [None] * CompiledFunction.metadata.num_backward_tokens
+
                 # - note: donated buffer logic requires (*ctx.symints, *ctx.saved_tensors) showing up first
                 #   in the bw output order.
+
+                # Every dereference of ctx.saved_tensors incurs saved_tensors_hooks calls
```

Comment:
ah good catch. This is also probably... the right thing to do anyway (we don't want to invoke the saved tensors hooks multiple times when someone uses them over compile)